cloudmarker/test/test_filestore.py

```diff
@@ -0,0 +1,69 @@
+"""Tests for filestore plugins."""
+
+
+import os
+import pathlib
+import shutil
+import unittest
+
+from cloudmarker.stores import filestore
+
+
+class FileStoreTest(unittest.TestCase):
+    """Tests for filestore plugins."""
+
+    @classmethod
+    def write_to_file(cls, filename):
+        # Initialize the FileStore plugin with the filename
+        store = filestore.FileStore(filename)
+
+        # dummy data to write
+        mp = {
+            "record_type": "dummy",
+            "data": {
+                "test_name": "tilde_filename"
+            }
+        }
+
+        store.write(mp)
+        store.done()
+
+    def test_filepath_tilde(self):
+        self.write_to_file("~/caramel_l33t/cloudmarker")
+        # if the parsing fails, it should create a "~" directory in the current
+        # path
+        f = pathlib.Path("~")
+
+        # If exists and is a directory, the parsing failed.
+        self.assertFalse(f.exists() and f.is_dir())
```

Comment:
I think only one of`f.is_dir()` and `f.exists()` should be enough. I don't see any specific advantage of using both checks. What do you think?

Reply:
I had used `f.exists()` and `f.is_dir()` because I did not see that errors are propagated. [Path.is_dir](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_dir) so there is no need to check for `exists()`, `is_dir()` implies `exists()`. I'll move to `is_dir()`.